### DISCUSSION TITLE
Include actionIdentifier in initialNotification

### DISCUSSION THREAD

**Discussion by TNicholson11** [invertase/react-native-firebase#4883] [Q&A]
When sending remote notifications with custom actions as described in a IOSNotificationCategory the [actionIdentifier](https://developer.apple.com/documentation/usernotifications/unnotificationresponse/1649548-actionidentifier) is a member of the [UNNotificationResponse](https://developer.apple.com/documentation/usernotifications/unnotificationresponse) however it is not included anywhere in the definition of a [FCM RemoteMessage](https://rnfirebase.io/reference/messaging/remotemessage) nor is it included in the _initialNotification returned by the [getInitialNotification method](https://rnfirebase.io/reference/messaging#getInitialNotification). Is there any particular motivation for this? Is there some limiting factor that I'm unaware of or have I totally overlooked some other solution for determining which action(s) were used to open a remotely triggered FCM notification?

Note: We are paying users of Notifee, however using data-only messages to populate Notifee notifications on the user-side is notoriously unreliable in background scenarios and, as far as I can tell, does NOT work whatsoever in low-power mode on iOS. However, remote FCM Notifications are delivered very reliably BUT without exposure to the actionIdentifier, custom action responses are not supported using FCM messaging alone.

It would be awesome if the actionIdentifier was included in the initialNotification and, if not, I'd be curious to know if there's another solution I have overlooked or if there is some reason that this feature cannot be supported.

Thanks very much!

**Comment by mikehardy** [comment]
@Ehesp or @Salakar do you know anything about this? I don't - my initial take @TNicholson11 is that it might simply have never been implement just because it wasn't thought of, without thinking farther than that. But I could be wrong! Hopefully Eliot or Mike will answer

**Reply by mikehardy** [reply]
I escalated internally to see if the original implementation authors might shed some light :-)

**Reply by TNicholson11** [reply]
Thank you @mikehardy 

I've updated RNFBMessaging+UNUserNotificationCenter.m locally to pass the actionIdentifier through as part of the notificationDict that is stored in _initialNotification in my testing with no issues so I'm hoping it's a simple enough addition. However, I haven't taken a look at what the equivalent on Android would be for parity.

We're migrating from the long since deprecated [react-native-fcm](https://github.com/evollu/react-native-fcm) so I'm just trying to get parity with that library. Unfortunately we have some time-sensitive / high-priority notifications (e.g. sms message, etc) so I think we will need to rely on notification payloads from FCM to ensure timely delivery (as opposed to using Notifee in all cases), but we'd love to get custom action support too!

Thanks again.

**Reply by mikehardy** [reply]
I share your thought that your change is likely fine, it's a PR I'd likely accept - and for the avoidance of doubt - would have a default stance of accepting if the original authors don't chime in. Parity between Android and iOS would be fantastic of course :-). And I totally understand, I also have lots of data-only / notifee action, but for a couple use cases I send a notification block myself, because mobile development is...challenging and rewarding.

**Reply by nabilfreeman** [reply]
Hi both. we have also reached this scenario and are displaying action buttons on the remote notifications for precisely the same reason as @TNicholson11, being the reliability of silent pushes in background and low power mode.

I've patched `RNFBMessaging+UNUserNotificationCenter.m` and got it working, so will open a PR with this.

**Reply by nabilfreeman** [reply]
Opened a PR here :)

https://github.com/invertase/react-native-firebase/pull/5473
